components/.storybook/main.js

```diff
@@ -7,5 +7,21 @@ module.exports = {
     "@storybook/addon-a11y",
     "@storybook/addon-actions",
     "storybook-addon-performance"
-  ]
+  ],
+  webpackFinal: async config => {
+    config.module.rules.push({
+      test: /\.(ts|tsx)$/,
+      use: [
+        {
+          loader: require.resolve("ts-loader")
```

Comment:
I normally recommend using Babel's typescript preset over `ts-loader`. That way, you can just stick to using `babel-loader`, just like if you were using vanilla JS. https://babeljs.io/docs/en/babel-preset-typescript

But it's just a suggestion. If you prefer `ts-loader`, then stick to that.